frontend/src/auth/LoginComponent.tsx

```diff
@@ -4,35 +4,48 @@ import { login } from "./auth-service";
 import LoginModel from "./login-model";
 import { setJwtHeader } from "../base-api";
 import { useNavigate } from "react-router-dom";
+import { useGlobalContext } from "./GlobalContext";
 
 const LoginComponent = () => {
-  const [username, setUsername] = useState("");
-  const [password, setPassword] = useState("");
+  const [usernameInput, setUsernameInput] = useState("");
+  const [passwordInput, setPasswordInput] = useState("");
   const [errorMessage, setErrorMessage] = useState("");
+  const { setLoggedIn, setMyUsername, setRoles } = useGlobalContext();
 
   const navigate = useNavigate();
 
+  const parseJwt = (token: string) => {
+    try {
+      return JSON.parse(atob(token.split(".")[1]));
+    } catch (err: any) {
+      return null;
+    }
+  };
+
   const handleLogin = async (e: any) => {
-    e.preventDefault();   
+    e.preventDefault();
 
-    if(username.length === 0 || password.length === 0)
+    if (usernameInput.length === 0 || passwordInput.length === 0)
       setErrorMessage("Please enter credentials");
 
     try {
       const loginModel: LoginModel = {
-        username,
-        password
+        username: usernameInput,
+        password: passwordInput,
       };
       const jwt = await login(loginModel);
+      const parsedJwt = parseJwt(jwt);
       setJwtHeader(jwt);
-      navigate(`/profile/${username}`);
-    }
-    catch (err: any) {
+      setLoggedIn(true);
+      setMyUsername(usernameInput);
```

Comment:
Username input is case insensitive. However, we need the case-sensitive username for myUsername (so we can perform conditional rendering on posts if your name matches the post author username)